Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1777,78 +1777,87 @@ extension BrowserViewController: WKNavigationDelegate {
             removeOpenInView()
         }
     }
-
-    private func openExternal(url: NSURL, prompt: Bool = true) {
-        if prompt {
-            // Ask the user if it's okay to open the url with UIApplication.
-            let alert = UIAlertController(
-                title: String(format: NSLocalizedString("Opening %@", comment:"Opening an external URL"), url),
-                message: NSLocalizedString("This will open in another application", comment: "Opening an external app"),
-                preferredStyle: UIAlertControllerStyle.Alert
-            )
-
-            alert.addAction(UIAlertAction(title: NSLocalizedString("Cancel", comment:"Alert Cancel Button"), style: UIAlertActionStyle.Cancel, handler: { (action: UIAlertAction) in
-            }))
-
-            alert.addAction(UIAlertAction(title: NSLocalizedString("OK", comment:"Alert OK Button"), style: UIAlertActionStyle.Default, handler: { (action: UIAlertAction!) in
-                UIApplication.sharedApplication().openURL(url)
-            }))
-
-            presentViewController(alert, animated: true, completion: nil)
-        } else {
-            UIApplication.sharedApplication().openURL(url)
+    
+    // Recognize an Apple Maps URL. This will trigger the native app. But only if a search query is present. Otherwise
```

Comment:
> `But only if a search query is present.`

That's not a particularly robust test since "a regular page on maps.apple.com" could easily contain a query, too. With some experimentation, I found that any query on the root `maps.apple.com` (e.g., http://maps.apple.com/?foo=bar) will be blank, yet any query with a path (e.g., https://www.apple.com/ios/maps/?foo=bar) will not. It's still hacky, but that means you could harden the query a bit more by also checking that the path is empty.